go/test/endtoend/recovery/pitr/main_test.go

```diff
@@ -0,0 +1,135 @@
+package pitr
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"vitess.io/vitess/go/test/endtoend/cluster"
+	"vitess.io/vitess/go/vt/log"
+)
+
+var (
+	clusterInstance *cluster.LocalProcessCluster
+	masterTablet    *cluster.Vttablet
+	replicaTablet   *cluster.Vttablet
+	initDBFile      string
+
+	cell                 = "zone1"
+	hostname             = "localhost"
+	binlogHost           = "127.0.0.1"
+	keyspaceName         = "ks"
+	partialRestoreKSName = "restoreks1"
+	fullRestoreKSName    = "restoreks2"
+	dbName               = "vt_ks"
+	shardName            = "0"
+	shardKsName          = "ks/0"
+	mysqlUserName        = "vt_dba"
+)
+
+func TestMain(m *testing.M) {
+	defer cluster.PanicHandler(nil)
+	flag.Parse()
+
+	exitCode, err := func() (int, error) {
+		clusterInstance = cluster.NewCluster(cell, hostname)
+		defer clusterInstance.Teardown()
+
+		// Start topo server
+		if err := clusterInstance.StartTopo(); err != nil {
+			log.Error(err)
+			return 1, err
+		}
+
+		if err := clusterInstance.VtctlProcess.CreateKeyspace(keyspaceName); err != nil {
+			log.Error(err)
+			return 1, err
+		}
+		clusterInstance.Keyspaces = append(clusterInstance.Keyspaces, cluster.Keyspace{Name: keyspaceName})
+
+		masterTablet = clusterInstance.NewVttabletInstance("replica", 0, cell)
+		replicaTablet = clusterInstance.NewVttabletInstance("replica", 0, cell)
+
+		err := startTablets([]*cluster.Vttablet{masterTablet, replicaTablet})
+		if err != nil {
+			return 1, err
+		}
+		err = clusterInstance.VtctlclientProcess.InitShardMaster(keyspaceName, shardName, cell, masterTablet.TabletUID)
+		if err != nil {
+			return 1, err
+		}
+		return m.Run(), nil
+	}()
+	if err != nil {
+		log.Error(err)
+		os.Exit(1)
+	}
+	os.Exit(exitCode)
+
+}
+
+func startTablets(tablets []*cluster.Vttablet) error {
+	var mysqlProcesses []*exec.Cmd
+	shard := &cluster.Shard{
+		Name: shardName,
+	}
+	for _, tablet := range tablets {
+		//tablet.MysqlctlProcess = cluster.MysqlCtlProcessInstance()
+		tablet.MysqlctlProcess = *cluster.MysqlCtlProcessInstance(tablet.TabletUID, tablet.MySQLPort, clusterInstance.TmpDirectory)
+		tablet.MysqlctlProcess.InitDBFile = initDBFile
+		if proc, err := tablet.MysqlctlProcess.StartProcess(); err != nil {
+			return err
+		} else {
+			// ignore golint warning, we need the else block to use proc
```

Comment:
don't you need a `//nolint` here? otherwise CI will fail.

Reply:
not sure why golangci-lint did not capture this earlier, it is passing without `//nolint`, to be on safe side I have added it.